rest/rest-client/src/test/java/functionaltests/SchedulerClientTest.java

```diff
@@ -315,6 +332,26 @@ public void testJobSubmissionEventListener() throws Exception {
     }
 
     @Test(timeout = MAX_WAIT_TIME)
+    public void testKillTask() throws Exception {
+        ISchedulerClient client = clientInstance();
+        Job job = createJob(NonTerminatingJob.class);
+        SchedulerEventListenerImpl listener = new SchedulerEventListenerImpl();
+        client.addEventListener(listener, true, SchedulerEvent.TASK_PENDING_TO_RUNNING);
+        JobId jobId = submitJob(job, client);
+
+        TaskInfo startedTask = listener.getStartedTask();
+        while (!startedTask.getJobId().value().equals(jobId.value())) {
+            startedTask = listener.getStartedTask();
```

Comment:
startedTask is a TaskInfo object, used in the loop to check that the task started corresponds to the job which was submitted. 

Basically the loop goes on until the task we want to kill is started